t/file.t

```diff
@@ -110,12 +110,31 @@ subtest "Regression Tests" => sub {
   # The empty string in turn caused URI::file->new_abs() to use the current
   # working directory as a default.
   {
-    my $sandbox_path = "/tmp/my sandbox";   # '%' in $good_uri causes the problem
-    my $file_uri     = URI::file->new_abs($sandbox_path);
+    my $test_loc     = $0 . ':' . __LINE__;
+    my $sandbox_unix = '/tmp/my sandbox';   # '%' in $good_uri causes the problem
+    my $sandbox_win  = '\tmp\my sandbox';
+    my $file_uri     = URI::file->new_abs($sandbox_unix);
     my $good_uri     = "file:///tmp/my%20sandbox";
 
     is( $file_uri,         $good_uri,     "file URI escaped correctly" );
-    is( $file_uri->file(), $sandbox_path, "got original filename (not current directory)" );
+
+    my $file_path = $file_uri->file();
+    $file_path    = '(undef)' unless defined $file_path;
+
+    if ( $file_path eq $sandbox_unix  or  $file_path eq $sandbox_win ) {
```

Comment:
Could we simplify this and say, just compare with the file path that `Path::Tiny` returns? It should already handle the path correctly for Windows.

Reply:
`Path::Tiny` would add another dependency (not core). Module `Cwd` could work, but it seems that this expression
already covers linux, windows, and mac? But what about all the other OS'? I've seen subtle differences like `/some/path` and `/some/path/`. Some kind of normalization would be needed but how to be sure w/o ever seeing a successful test-run on a non-common OS (see `perldoc perlport`)?
One could add a long random string to the test-path and if that is missing in the result, it might be the current working directory. This might cover all OS, but the idea here was: If it looks like some proven valid result, it is good. If it returns the same as the default - which should be identical independently of the OS if it is computed the same way `URI` does - it's the bug again. Otherwise, we don't know and should ask for a new test case. Though, we cannot call it a failed test unless we know what the correct result would have been.
This way, we could also catch deviations from the expected bug, although it was not our intention.
The outcome  is either 100% correct, it is 100%  wrong, or we don't know for sure. 
Line 121 is required. Line 122 is a portable way to assure a defined value (i.e. w/o `//=`; the table at the top of `t/file.t` indicates that `undef` is a potential result). Line 124 is okay for now, but could be a hash lookup in case more valid results (VMX?) need to be compared. I would say, leave it as it is.

Reply:
It's only a test, so Path::Tiny could be added as a test/recommends dep and the test could be optionally run via Test::Needs.

Reply:
Sure, but I see no gain in comparing the exact path. There might be differences and I have no access to other OS' to validate the test itself. 
The test result in this case would be some path if the bug is present.
However (after some pondering), it boils down to the question - is the value returned identical to the default value? Which is the path, independent of the OS or how the path is represented. Any deviation would be a bug, independent of the purpose of this specific test (vulnerability). This would be a simplification and we would also get rid of storing potentially valid results (line 124). Maintaining valid paths for the more exotic OS' are no longer needed then.

So, this test could be replaced and broadened by just testing if the URI of some path that is very very unlikely an existing path is identical to what `URI::file->new_abs()->file()` returns (then it is a bug).  ... unless I missed something.

Reply:
I would prefer any test of this sort to be as simple as possible, not introducing any unnecessary complexity and not introducing portability issues. So, if a simple check `URI::file->new_abs->file` does the trick, that sounds good.